DiaHelperApplication/DiaHelperApplication/Controllers/HomeViewController.swift

```diff
@@ -31,34 +56,45 @@ class HomeViewController: UIViewController{
         return button
     }()
     
-    private func setupReminderButton() {
-        let color = UIColor(named: "newBrown")
-        
-        let textAttributes: [NSAttributedString.Key: Any] = [
-                .foregroundColor: color as Any,
-                .font: UIFont.systemFont(ofSize: 16)
-            ]
-        
-        let reminderBarButton = UIBarButtonItem(title: "Reminder", style: .plain, target: self, action: #selector(didTapReminder))
-        
-        reminderBarButton.setTitleTextAttributes(textAttributes, for: .normal)
+    public var glucometerButton: UIButton = {
+        let button = UIButton(type: .system)
+        button.setImage(UIImage(named: "glucometerImage")?.withRenderingMode(.alwaysOriginal), for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
         
-        navigationItem.rightBarButtonItem = reminderBarButton
-    }
-    
-    @objc private func didTapReminder(_ action: UIAction) {
-        let addNutritionViewController = AddNutritionViewController()
-        let navController = UINavigationController(rootViewController: addNutritionViewController)
-        navController.modalPresentationStyle = .fullScreen
-        navigationController?.present(navController, animated: true)
-    }
-
+        return button
+    }()
 
     override func viewDidLoad() {
         super.viewDidLoad()
         view.backgroundColor = .systemGray6
+        self.navigationController?.isNavigationBarHidden = true
+        
+        if let user = UserManager.shared.getCurrentUser() {
+            currentUser = user
+            
+            if let meals = currentUser?.meals, !meals.isEmpty {
+                self.meals = meals
+                tableView.reloadData()
+            }
+        }
+        
+        setupTimer()
         setupAddButton()
-        setupReminderButton()
+        glucometerAction()
+        addTestMeals()
+        setupTableView()
+        addSubviews()
+        addViewConstraints()
+    }
+    
+    private func setupTimer() {
+        if let user = currentUser {
+            if user.nightscout.isEmpty{
+                self.bloodSugar.text = "-"
+                return
+            }
+        }
+        //updateLabel()
         
         timer = Timer.scheduledTimer(timeInterval: 300.0, // 300 секунди = 5 минути
```

Comment:
Here it works and is not a problem as updateLabel will eventually be called, but you should be aware that the Timer is imprecise - there is no guarantee it will be 300.0 seconds and it pauses when the app goest to the background and resumes when it comes back to the foreground.

Reply:
А как по друг начин би могло да се извиква през дадено време функцията?